lib/llm/src/block_manager/block/registry.rs

```diff
@@ -41,7 +41,7 @@ pub struct UnregisterFailure(SequenceHash);
 
 #[derive()]
 pub struct BlockRegistry {
-    blocks: HashMap<SequenceHash, Weak<RegistrationHandle>>,
+    blocks: HashMap<SequenceHash, Weak<()>>,
```

Comment:
It took me a bit to get here.  this is one module we haven't added comprehensive docstrings.

could you please add module level (//!) write up explaining how the `Arc<()>` is the storage specific block_handle, which is always held with an `Arc<RegistrationHandle>`, thus the liveness of one is directly correlated with the other.